lib/chef/knife/bootstrap.rb

```diff
@@ -432,6 +434,22 @@ def validate_options!
         true
       end
 
+      # Ensure options are valid by checking bootstrap version values.
+      #
+      # The method call will cause the program to exit(1) if:
+      #   * Bootstrap version is greater than or equal to 15 for chef 14 or lower
+      #   * Pre-release options is passed for chef 14 or lower
+      #
+      # @return [TrueClass] If options are valid.
+      def validate_bootstrap_version_options!
+        if target_node_gt_15?
```

Comment:
this would be more understandable if both the conditions were here, e.g.

```
if (Chef::VERSION.to_i < 15) && target_version_gte_15?
```

note that since we're putting this on the Chef 14 branch, in theory Chef::VERSION.to_i should always be < 15, but I guess it's safe to be clear.